app/src/main/java/org/oppia/app/topic/train/TopicTrainFragmentPresenter.kt

```diff
@@ -4,20 +4,61 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.LinearLayoutManager
 import org.oppia.app.databinding.TopicTrainFragmentBinding
 import org.oppia.app.fragment.FragmentScope
+import org.oppia.app.viewmodel.ViewModelProvider
 import javax.inject.Inject
 
 /** The presenter for [TopicTrainFragment]. */
 @FragmentScope
 class TopicTrainFragmentPresenter @Inject constructor(
-  private val fragment: Fragment
-) {
+  private val fragment: Fragment,
+  private val viewModelProvider: ViewModelProvider<TopicTrainViewModel>
+) : SkillInterface {
+  private val selectedSkillList = ArrayList<String>()
+
   fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View? {
+    val skillAdapter = SkillSelectionAdapter(dummySkillList(), this)
+
     val binding = TopicTrainFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false)
+    binding.skillRecyclerView.apply {
+      adapter = skillAdapter
+      layoutManager = LinearLayoutManager(context)
+    }
     binding.let {
+      it.viewModel = getTopicTrainViewModel()
       it.lifecycleOwner = fragment
     }
     return binding.root
   }
+
+  private fun getTopicTrainViewModel(): TopicTrainViewModel {
+    return viewModelProvider.getForFragment(fragment, TopicTrainViewModel::class.java)
+  }
+
+  override fun skillSelected(skill: String) {
+    selectedSkillList.add(skill)
+    getTopicTrainViewModel().selectedSkillList(selectedSkillList)
+  }
+
+  override fun skillUnselected(skill: String) {
+    selectedSkillList.remove(skill)
+    getTopicTrainViewModel().selectedSkillList(selectedSkillList)
+  }
+
+  private fun dummySkillList(): List<String> {
```

Comment:
Can we fetch these from a domain controller, instead? E.g. https://github.com/oppia/oppia-android/blob/develop/domain/src/main/java/org/oppia/domain/topic/TopicController.kt#L33 & then retrieve the skill list. Presumably this fragment needs to be initialized with a topic ID which seems like an important part of the arguments API for the fragment that should be introduced with its initial structure.

Reply:
Done. Started fetching skills from domain layer. This implementation might get updated once tabs are introduced in #135